include/alpaka/mem/view/Accessor.hpp

```diff
@@ -0,0 +1,195 @@
+
+#pragma once
+
+#include <alpaka/dim/Traits.hpp>
+#include <alpaka/mem/view/Traits.hpp>
+
+namespace alpaka
+{
+    template<typename Pointer, typename Value, typename Idx, typename Dim>
+    struct Accessor;
+
+    template<typename Pointer, typename Value, typename Idx>
+    struct Accessor<Pointer, Value, Idx, DimInt<1>>
+    {
+        auto& operator[](Vec<DimInt<1>, Idx> i) const
+        {
+            return (*this)(i[0]);
+        }
+
+        auto& operator[](Idx i) const
+        {
+            return (*this)(i);
+        }
+
+        auto& operator()(Idx i) const
+        {
+            return p[i];
+        }
+
+        Pointer p;
+        Vec<DimInt<1>, Idx> extents;
+    };
+
+    template<typename Pointer, typename Value, typename Idx>
+    struct Accessor<Pointer, Value, Idx, DimInt<2>>
+    {
+        auto& operator[](Vec<DimInt<2>, Idx> i) const
+        {
+            return (*this)(i[0], i[1]);
+        }
+
+        auto& operator()(Idx y, Idx x) const
+        {
+            return *(reinterpret_cast<Pointer>(reinterpret_cast<char*>(p) + y * rowPitchInBytes) + x);
+        }
+
+        Pointer p;
+        Idx rowPitchInBytes;
+        Vec<DimInt<2>, Idx> extents;
+    };
+
+    template<typename Pointer, typename Value, typename Idx>
+    struct Accessor<Pointer, Value, Idx, DimInt<3>>
+    {
+        auto& operator[](Vec<DimInt<3>, Idx> i) const
+        {
+            return (*this)(i[0], i[1], i[2]);
+        }
+
+        auto& operator()(Idx z, Idx y, Idx x) const
+        {
+            return *(
+                reinterpret_cast<Pointer>(reinterpret_cast<char*>(p) + z * slicePitchInBytes + y * rowPitchInBytes)
+                + x);
+        }
+
+        Pointer p;
+        Idx rowPitchInBytes;
+        Idx slicePitchInBytes;
+        Vec<DimInt<3>, Idx> extents;
+    };
+
+#if 0
+    using Image = cudaTextureObject_t;
+
+    template<typename Value, typename Idx>
+    struct Accessor<Image, Value, Idx, DimInt<1>>
+    {
+        auto operator[](Vec<DimInt<1>, Idx> i) const -> Value
+        {
+            return (*this)(i[0]);
+        }
+
+        auto operator[](Idx i) const -> Value
+        {
+            return (*this)(i);
+        }
+
+        auto operator()(Idx i) const -> Value
+        {
+            return tex1Dfetch(texObj, i);
+        }
+
+        auto operator()(float i) const -> Value
+        {
+            return tex1D(texObj, i);
+        }
+
+        Image texObj;
+        Vec<DimInt<1>, Idx> extents;
+    };
+
+    template<typename Value, typename Idx>
+    struct Accessor<Image, Value, Idx, DimInt<2>>
+    {
+        auto operator[](Vec<DimInt<2>, Idx> i) const -> Value
+        {
+            return (*this)(i[0], i[1]);
+        }
+
+        auto operator()(Idx y, Idx x) const -> Value
+        {
+            return tex1Dfetch(texObj, y * rowPitchInValues + x);
+        }
+
+        auto operator()(float y, float x) const -> Value
+        {
+            return tex2D(texObj, x, y);
+        }
```

Comment:
We can always support integral access to the texels using `tex1Dfetch` on ND textures. For floating point coordinates we can use the corresponding `texND` functions. Also notice the all functions return by value, because we cannot form references into texture memory.